### PR TITLE
Developer guide: @SqlQuery and @SqlBatch

### PR DESCRIPTION
**Pull request by qualidafial**

### CODE REVIEW

**Comment by qualidafial** [comment]
@stevenschlansker or @arteam: Could one of you please review this?

File: docs/src/adoc/index.adoc [hunk 1]
```diff
@@ -1613,47 +1620,351 @@ As in the Core API, neither `@BindMap`, `@BindBean`, `@BindFields`, nor
...
+method return type looks like a collection.
+
+[source,java]
+----
+public interface UserDao {
+  @SqlQuery("select name from users")
+  List<String> listNames(); // <1>
+
+  @SqlQuery("select name from users where id = ?")
+  String getName(long id); // <2> <3>
```
Comment (arteam) [inline docs/src/adoc/index.adoc]: Does it makes sense to mention that JDBI supports out-of-box `Optional<String>` for single row results?

File: sqlobject/src/test/java/org/jdbi/v3/sqlobject/TestRegisteredMappersWork.java [hunk 2]
```diff
@@ -102,6 +103,25 @@ public void setColor(String color)
...
+
+        BeanMappingDao dao = jdbi.onDemand(BeanMappingDao.class);
+        dao.useHandle(h -> {
+            TestColumnMappers.SomeBeanDao another = h.attach(TestColumnMappers.SomeBeanDao.class);
+            Bean foo = dao.findByName("foo");
+
+            foo.setColor("brown");
+            dao.insertBean(foo);
+        });
+    }
```
Comment (qualidafial) [inline sqlobject/src/test/java/org/jdbi/v3/sqlobject/TestRegisteredMappersWork.java]: Whoops, I did not mean to check this in.. was showing a coworker the correct way to negotiate on-demand with shared handles.

Comment (qualidafial) [reply sqlobject/src/test/java/org/jdbi/v3/sqlobject/TestRegisteredMappersWork.java]: Will remove this in a follow up PR for other docs.

**Pull request merged**